generators/pythonv2/codegen/src/ast/PythonFile.ts

```diff
@@ -0,0 +1,103 @@
+import { AstNode } from "./core/AstNode";
+import { Writer } from "./core/Writer";
+import { Class } from "./Class";
+import { Method } from "./Method";
+import { CodeBlock } from "./CodeBlock";
+import { Reference } from "./Reference";
+
+export declare namespace PythonFile {
+    interface Args {
+        /* The name of the Python module that this file belongs to*/
+        moduleName: string;
+        /* The path of the Python file relative to the module */
+        path: string[];
+        /* The name of the Python file */
+        name: string;
+    }
+}
+
+export class PythonFile extends AstNode {
+    public readonly moduleName: string;
+    public readonly path: string[];
+    public readonly name: string;
+    private readonly statements: (Class | Method | CodeBlock)[] = [];
+
+    constructor({ moduleName, path, name }: PythonFile.Args) {
+        super();
+        this.moduleName = moduleName;
+        this.path = path;
+        this.name = name;
+    }
+
+    public addStatement(statement: Class | Method | CodeBlock): void {
+        this.statements.push(statement);
+    }
+
+    public write(writer: Writer): void {
+        this.writeImports(writer);
+        this.statements.forEach((statement, idx) => {
+            statement.write(writer);
+            writer.newLine();
+            if (idx < this.statements.length - 1) {
+                writer.newLine();
+            }
+        });
+    }
+
+    /*******************************
+     * Helper Methods
+     *******************************/
+
+    private writeImports(writer: Writer): void {
+        const references = writer.getReferences();
```

Comment:
I found it more convenient for the writer to still be responsible for tracking references. I could see this living separately in a "context" object or similar, but keeping it simple by using the writer as the context for now. It felt heavy-handed to try to use the `PythonFile` node to keep track of all references.